### ISSUE TITLE
support for multi-tenant mongo

### ISSUE THREAD

**Issue by saahn** [StackStorm/st2#2670]
It looks like mongoengine requires connection registration with aliases in order to switch between multiple databases.

Our custom pack also uses mongo as a backend database, and it took us a while to figure out why our code was defaulting to the `st2` database even though we were explicitly specifying our own database in our code where we establish mongoengine connection (e.g. `me_client = me.connect('foo', host='127.0.0.1', port='27017')`.

I followed the workaround described in this [issue](https://github.com/MongoEngine/mongoengine/issues/604), i.e. also added `        me.register_connection(alias='foo', name='foo', ...)
` where we were creating our mongoengine connection.

This is a limitation of mongoengine, but I think the stackstorm codebase should be registering an alias as well to prevent `st2` from getting set as the default database. Probably just a one-line change in https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/db/__init__.py#L62

**Comment by manasdk** [comment]
Hmm, this seems very reasonable. Let me investigate and figure this out further.

Thanks for the report!

**Comment by manasdk** [comment]
@saahn I looked at the code and tried to see if willing mongoengine to do this part is possible. As a one time thing I discovered that changing the code should be possible however going forward it will become a bit of a maintenance nightmare for us since everytime a new model is added we would need to make sure that `meta = {"db_alias": "st2"}` would have to be properly assigned. We could build out some abstractions that help us take this route but then I cannot truly answer the question that st2 owning the default connection is an issue.

At this point I am reluctant to change anything here unless we can bite the bullet and build out suitable abstractions that make it possible to maintain. Essentially, anything that needs to be specially remembered while adding new models is a potential source of bugs.
